epochStart/metachain/validators.go

```diff
@@ -194,8 +235,116 @@ func (vic *validatorInfoCreator) VerifyValidatorInfoMiniBlocks(
 	return nil
 }
 
-// SaveValidatorInfoBlocksToStorage saves created data to storage
-func (vic *validatorInfoCreator) SaveValidatorInfoBlocksToStorage(_ data.HeaderHandler, body *block.Body) {
+// GetLocalValidatorInfoCache returns the local validator info cache which holds all the validator info for the current epoch
+func (vic *validatorInfoCreator) GetLocalValidatorInfoCache() epochStart.ValidatorInfoCacher {
+	return vic.dataPool.CurrentEpochValidatorInfo()
+}
+
+// CreateMarshalledData creates the marshalled data to be sent to shards
+func (vic *validatorInfoCreator) CreateMarshalledData(body *block.Body) map[string][][]byte {
+	if !vic.enableEpochsHandler.IsRefactorPeersMiniBlocksFlagEnabled() {
+		return nil
+	}
+
+	if check.IfNil(body) {
+		return nil
+	}
+
+	marshalledValidatorInfoTxs := make(map[string][][]byte)
```

Comment:
It is done like this to be easily used in the caller side as there the rewardsCreator returns a map (because there the topic could be different depending of the receiver shard) 

`broadcastTopic := createBroadcastTopic(factory.RewardsTransactionTopic, brc.shardCoordinator, miniBlock.ReceiverShardID)`

and together with the validatorInfoCreator all the txs should be broadcast one time.

The caller side code:

```
func (mp *metaProcessor) getAllMarshalledTxs(body *block.Body) map[string][][]byte {
	allMarshalledTxs := make(map[string][][]byte)

	marshalledRewardsTxs := mp.epochRewardsCreator.CreateMarshalledData(body)
	marshalledValidatorInfoTxs := mp.validatorInfoCreator.CreateMarshalledData(body)

	for topic, marshalledTxs := range marshalledRewardsTxs {
		allMarshalledTxs[topic] = append(allMarshalledTxs[topic], marshalledTxs...)
		log.Trace("metaProcessor.getAllMarshalledTxs", "topic", topic, "num rewards txs", len(marshalledTxs))
	}

	for topic, marshalledTxs := range marshalledValidatorInfoTxs {
		allMarshalledTxs[topic] = append(allMarshalledTxs[topic], marshalledTxs...)
		log.Trace("metaProcessor.getAllMarshalledTxs", "topic", topic, "num validator info txs", len(marshalledTxs))
	}

	return allMarshalledTxs
}

```

Reply:
And if in the future we will have also for validator info different topics, just to send different information depending of the receiver shard, the current implementation will work as it is with minimal changes.

Reply:
Anyway I have refactored the code but I kept the return type